base/compiler/optimize.jl

```diff
@@ -828,9 +832,9 @@ function inline_worthy(ir::IRCode,
         stmt = ir.stmts[line][:inst]
         thiscost = statement_or_branch_cost(stmt, line, ir, ir.sptypes, union_penalties, params)
         bodycost = plus_saturate(bodycost, thiscost)
-        bodycost > cost_threshold && return false
+        bodycost > cost_threshold && return MAX_INLINE_COST
     end
-    return true
+    return bodycost
```

Comment:
```suggestion
    return clamp(bodycost, UInt16)
```

Reply:
It seems we don't have access to `clamp` at this stage of bootstrapping, so I just wrote a function that constrains an `Int` between `0x0010` and `typemax(UInt16)`